internal/route/node_filter.go

```diff
@@ -107,6 +107,32 @@ func (f *IsDoneSyncing) Apply(_ metadata.RequestMetadata, upstreamConfig *config
 	return true
 }
 
+type IsLatencyAcceptable struct {
```

Comment:
Is this type only for checking against the passive latency checks? 

I'm starting to get the feeling that keeping around all this code for the passive checks might be unnecessary and actually be causing more harm than good. It's starting to feel confusing.

WDYT about only keeping around the absolute minimal amount of code around for passive health checks (e.g. keep around a few methods, leave out the wiring/glue).